mig-agent/config.go

```diff
@@ -52,54 +52,205 @@ func configLoad(path string) (err error) {
-	if err != nil {
-		panic(err)
+	return
+}
+
+// Globals receives parsed config settings and applies them to global vars.
+// NewGlobals returns a Globals struct populated with initial values from global vars.
+type Globals struct {
+
+	// restart the agent on failures, don't let it die
+	ISIMMORTAL bool
+
+	// request installing of a service to start the agent at boot
+	MUSTINSTALLSERVICE bool
+
+	// attempt to discover the public IP of the endpoint by querying the api
+	DISCOVERPUBLICIP bool
+
+	// attempt to discover meta-data for instances running in AWS
+	DISCOVERAWSMETA bool
+
+	// in check-in mode, the agent connects to the relay, runs all pending commands
+	// and exits. this mode is used to run the agent as a cron job, not a daemon.
+	CHECKIN bool
+
+	// if enabled, the agent will inform modules to mask returned meta-data as much
+	// as possible. modules which consider this will tell you they found something,
+	// but not much else.
+	EXTRAPRIVACYMODE bool
+
+	// spawn persistent modules; if enabled in the built-in config this can be
+	// disabled at run-time using a config option or command line flag
+	SPAWNPERSISTENT bool
+
+	// how often the agent will refresh its environment. if 0 agent
+	// will only update environment at initialization.
+	REFRESHENV time.Duration
+
+	LOGGINGCONF mig.Logging
+
+	// location of the rabbitmq server
+	// if a direct connection fails, the agent will look for the environment
+	// variables HTTP_PROXY and HTTPS_PROXY, and retry the connection using
+	// HTTP CONNECT proxy tunneling
+	AMQPBROKER string
+
+	// location of the MIG API, used for discovering the public IP
+	APIURL string
+
+	// if the connection still fails after looking for a HTTP_PROXY, try to use the
+	// proxies listed below
+	PROXIES []string
+	// If you don't want proxies in the built-in configuration, use the following
+	// instead.
+	// var PROXIES = []string{}
+
+	// local socket used to retrieve stat information from a running agent
+	SOCKET string
+
+	// frequency at which the agent sends heartbeat messages
+	HEARTBEATFREQ time.Duration
+
+	// timeout after which a module run is killed
+	MODULETIMEOUT time.Duration
+
+	// Not supported by config
+	// Control modules permissions by PGP keys
+	// AGENTACL [...]string
+
+	// Not supported by config
+	// PGP public keys that are authorized to sign actions
+	// this is an array of strings, put each public key block
+	// into its own array entry, as shown below
+	// PUBLICPGPKEYS [...]string
+
+	// CA cert that signs the rabbitmq server certificate, for verification
+	// of the chain of trust. If rabbitmq uses a self-signed cert, add this
+	// cert below
+	CACERT []byte
+
+	// All clients share a single X509 certificate, for TLS auth on the
+	// rabbitmq server. Add the public client cert below.
+	AGENTCERT []byte
+
+	// Add the private client key below.
+	AGENTKEY []byte
+}
+
+func NewGlobals() *Globals {
+	return &Globals{
+		ISIMMORTAL:         ISIMMORTAL,
```

Comment:
If we decide to only use this struct temporarily then we can convert, otherwise if we use this to replace globals then keep them exported.